vidarr/wdl.py

```diff
@@ -244,72 +241,56 @@ def read_output(output: WDL.Decl):
-        'language': 'WDL_' + str(
-            doc.wdl_version).replace(
-            ".",
-            "_"),
-        'outputs': {
-            workflow_name + "." + output.name:
-                read_output(output)
-            for output in doc.workflow.outputs},
-        'parameters': workflow_inputs,
-        'workflow': doc.source_text,
-        'accessoryFiles': {
-            imported.uri: imported.doc.source_text for imported in doc.imports}}
-    
     output_meta = doc.workflow.meta.get("output_meta", {})
 
-    output_lines = []
-
+    wdl_doc = doc.source_lines
+    add_empty_optional_pair_for_vidarr_labels = False
     # Iterate over each output defined in the workflow
     for output in doc.workflow.outputs:
 
-        output_line = str(output)
-
         # Check if the output name exists in the output metadata
         if output.name in output_meta:
             output_metadata = output_meta[output.name]
 
             if isinstance(output_metadata, dict) and 'vidarr_label' in output_metadata:
                 vidarr_label = output_metadata['vidarr_label']
 
-                if isinstance(output.type, WDL.Type.File) and not output.type.optional:
-
-                    # Construct the modified output line
-                    output_line = f"   Pair[File, Map[String, String]] {output.name} = ({output.expr}, {{\"vidarr_label\": \"{vidarr_label}\"}})"
-                    
-                elif isinstance(output.type, WDL.Type.Pair) and \
-                    isinstance(output.type.left_type, WDL.Type.File) and \
-                    isinstance(output.type.right_type, WDL.Type.Map):
-
-                    output_line = f"    Pair[File, Map[String, String]] {output.name} = {output.expr}"
+                source_line = output.pos.line - 1
+                start_position = output.pos.column - 1
+                end_position = output.pos.end_column - 1
 
-        # Append output lines to a consistently updated list
-        output_lines.append(output_line)
-
-    # Define the pattern for searching and replacing the output block
-    pattern = r"(?:workflow)([\s\S]*?)(?:output\s*{)([\s\S]*?)(?:\}\s*\n)"
-
-    # Search for the pattern in the workflow
-    match = re.search(pattern, workflow['workflow'], re.DOTALL)
-
-    # If the pattern is found, perform the replacement
-    if match:
-
-        # Extracting the part of the workflow string where the output block is located
-        output_block_start = match.start(2)
-        output_block_end = match.end(2)
-        output_block_text = match.group(2)
+                if isinstance(output.type, WDL.Type.File) and not output.type.optional:
+                    output_line = (f'Pair[File, Map[String, String]] {output.name} = '
+                                   f'({output.expr}, {{"vidarr_label": "{vidarr_label}"}})')
+                elif isinstance(output.type, WDL.Type.File) and output.type.optional:
+                    add_empty_optional_pair_for_vidarr_labels = True
```

Comment:
Why is this always True in this case?

Reply:
If we're working with a wdl that has vidarr_label and `File?`, we need to create a `Pair[File, Map[String,String]]?` empty_optional_pair.

Reply:
If we're working with a wdl that has vidarr_label and `File?`, then why are we adding an _empty_ optional pair? There might be a File there?

Reply:
Added some comments to clarify, but more directly, if we have an output file return Pair[...] else return Pair[...]? - returning Pair[...]? is to ensure the final return type is what we expect.